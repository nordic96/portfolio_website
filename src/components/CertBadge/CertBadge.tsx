import React from 'react';

export interface CertBadgeProps {
    href: string;
    src: string;
    spanClassName?: string;
    title: string;
    desc: string;
}

const CertBadge = (props: CertBadgeProps) => {
    const { src, href, spanClassName, title, desc } = props;
    return (
        <div
            className={
                'flex flex-column radius-1 items-center gap-1 w-32 max-sm:w-24'
            }>
            <p className={'flex rounded-full bg-sky-900 px-2 text-white'}>
                {desc}
            </p>
            <a href={href} target={'_blank'} rel="noreferrer">
                <span className={spanClassName}>
                    <img
                        alt={'cert-badge'}
                        className={'w-full hover:motion-safe:animate-bounce'}
                        src={src}
                    />
                </span>
                <p className={'text-sky-900 font-semibold text-base'}>
                    {title}
                </p>
            </a>
        </div>
    );
};

export default CertBadge;
