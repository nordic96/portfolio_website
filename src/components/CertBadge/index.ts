import CertBadge from './CertBadge';

export default CertBadge;
