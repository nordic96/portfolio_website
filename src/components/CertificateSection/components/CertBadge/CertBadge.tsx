import React from 'react';
import { CertificateDTO } from '../../../../models/cetificate/Certificate';

const COLOR_MAP: Record<string, string> = {
    violet: 'shadow-violet-500/50',
    slate: 'shadow-slate-500/50',
    indigo: 'shadow-indigo-500/50',
    teel: 'shadow-teel-500/50',
    default: '',
};

const CertBadge: React.FC<CertificateDTO> = (props: CertificateDTO) => {
    const { theme_color, name, credentials_url, year_obtained, logo_src } =
        props;
    let baseClass =
        'flex flex-col rounded-md radius-1 shadow-md text-center dark:bg-gray-800 px-2 py-1 max-sm:px-1 items-center gap-1 w-36 max-sm:w-32 text-black dark:text-white ';
    baseClass = baseClass.concat(COLOR_MAP[theme_color]);

    return (
        <a
            data-testid={'credentials_url'}
            href={credentials_url}
            target={'_blank'}
            rel="noreferrer">
            <div data-testid={'badge_container'} className={baseClass}>
                <p>{year_obtained}</p>
                <span className={`rounded-full bg-radial backdrop-opacity-10`}>
                    <img
                        alt={'cert-badge'}
                        className={
                            'h-28 max-sm:h-20 motion-safe:hover:animate-bounce'
                        }
                        src={logo_src}
                    />
                </span>
                <p className={'text-base font-semibold uppercase'}>{name}</p>
            </div>
        </a>
    );
};

export default CertBadge;
