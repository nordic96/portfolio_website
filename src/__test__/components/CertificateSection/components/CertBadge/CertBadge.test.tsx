import React from 'react';
import { render, screen } from '@testing-library/react';
import CertBadge from '../../../../../components/CertificateSection/components/CertBadge';

describe('CertBadge Component', () => {
    it('should render cert badge component with correct labels', () => {
        render(
            <CertBadge
                logo_src="/test"
                name="name_test"
                theme_color="violet"
                credentials_url="url_test"
                year_obtained="test_2025"
            />
        );
        expect(screen.getByText('name_test')).toBeInTheDocument();
        expect(screen.getByTestId('credentials_url')).toHaveAttribute(
            'href',
            'url_test'
        );
        expect(screen.getByText('test_2025')).toBeInTheDocument();
        const divComponent = screen.getByTestId('badge_container');
        expect(divComponent.classList).toContain('shadow-violet-500/50');
    });

    it('should match snapshot', () => {
        const comp = render(
            <CertBadge
                logo_src="/test"
                name="name_test"
                theme_color="violet"
                credentials_url="url_test"
                year_obtained="test_2025"
            />
        );
        expect(comp).toMatchSnapshot();
    });
});
