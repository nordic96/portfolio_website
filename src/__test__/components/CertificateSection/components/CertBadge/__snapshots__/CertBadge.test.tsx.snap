// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertBadge Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        data-testid="credentials_url"
        href="url_test"
        rel="noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col rounded-md radius-1 shadow-md text-center dark:bg-gray-800 px-2 py-1 max-sm:px-1 items-center gap-1 w-36 max-sm:w-32 text-black dark:text-white shadow-violet-500/50"
          data-testid="badge_container"
        >
          <p>
            test_2025
          </p>
          <span
            class="rounded-full bg-radial backdrop-opacity-10"
          >
            <img
              alt="cert-badge"
              class="h-28 max-sm:h-20 motion-safe:hover:animate-bounce"
              src="/test"
            />
          </span>
          <p
            class="text-base font-semibold uppercase"
          >
            name_test
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      data-testid="credentials_url"
      href="url_test"
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="flex flex-col rounded-md radius-1 shadow-md text-center dark:bg-gray-800 px-2 py-1 max-sm:px-1 items-center gap-1 w-36 max-sm:w-32 text-black dark:text-white shadow-violet-500/50"
        data-testid="badge_container"
      >
        <p>
          test_2025
        </p>
        <span
          class="rounded-full bg-radial backdrop-opacity-10"
        >
          <img
            alt="cert-badge"
            class="h-28 max-sm:h-20 motion-safe:hover:animate-bounce"
            src="/test"
          />
        </span>
        <p
          class="text-base font-semibold uppercase"
        >
          name_test
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
